src/modules/playground.ts

```diff
@@ -76,6 +88,30 @@ export class PlaygroundModule extends Module {
 		}
 	}
 
+	@listener({ event: 'message' })
+	async onPGLinkAttachment(msg: Message) {
+		const attachment = msg.attachments.find(a => a.name === 'message.txt');
+		if (!attachment) return;
+		const content = await fetch(attachment.url).then(r => r.text());
+		await this.onLongPGLink(msg, content, !!msg.content, true);
+	}
+
+	private async shortenPGLink(url: string) {
+		const response = await fetch(LINK_SHORTENER_ENDPOINT, {
+			method: 'post',
+			body: JSON.stringify({ url }),
```

Comment:
- `createdOn` is used for stats, it's optional but since this uses the API I'd like to keep track of that
- `expires` isn't yet used but at some point I'll clean up links that are old and haven't been visited. Since the history is persisted on Discord it's better to not expire those links either.

```suggestion
			body: JSON.stringify({ url, createdOn: "api", expires: false }),
```

Reply:
Maybe make that `createdOn: "discord-bot"`? Then you know the exact source?

Reply:
Makes sense 👌

Now only `api` `client` `plugin` and `other` are supported.

To avoid blocking the PR we can merge it with "api" and once I add support for `discord-bot` I'll open a PR.

Reply:
gillchristian/tsplay.dev#69